storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/dataaccess/V3RocksDbDao.java

```diff
@@ -246,8 +246,16 @@ public void addHotBlocks(final Map<Bytes32, SignedBeaconBlock> blocks) {
     }
 
     @Override
-    public void addHotStateRoot(final Bytes32 stateRoot, final SlotAndBlockRoot slotAndBlockRoot) {
-      transaction.put(V3Schema.STATE_ROOT_TO_SLOT_AND_BLOCK_ROOT, stateRoot, slotAndBlockRoot);
+    public void addHotStateRoots(
+        final Map<Bytes32, SlotAndBlockRoot> stateRootToSlotAndBlockRootMap) {
+      stateRootToSlotAndBlockRootMap
+          .keySet()
+          .forEach(
+              stateRoot ->
+                  transaction.put(
+                      V3Schema.STATE_ROOT_TO_SLOT_AND_BLOCK_ROOT,
+                      stateRoot,
+                      stateRootToSlotAndBlockRootMap.get(stateRoot)));
```

Comment:
Can just use a `forEach` directly on the map:
```suggestion
      stateRootToSlotAndBlockRootMap
          .forEach(
              (stateRoot, blockRoot) ->
                  transaction.put(
                      V3Schema.STATE_ROOT_TO_SLOT_AND_BLOCK_ROOT,
                      stateRoot,
                      blockRoot));
```